File: source/dotnet/Library/AdaptiveCards.Rendering.Wpf/AdaptiveMediaEventArgs.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
using System;

namespace AdaptiveCards.Rendering.Wpf
{
    public class AdaptiveMediaEventArgs : EventArgs
    {
        public AdaptiveMediaEventArgs(AdaptiveMedia media)
        {
            Media = media;
        }

        /// <summary>
        /// The clicked media
        /// </summary>
        public AdaptiveMedia Media { get; set; }
    }
}

File: source/winappsdk/Renderer/lib/AdaptiveMediaEventArgs.cpp
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
#include "pch.h"
#include "AdaptiveMediaEventArgs.h"

using namespace Microsoft::WRL;
using namespace ABI::AdaptiveCards::Rendering::WinUI3;
using namespace ABI::AdaptiveCards::ObjectModel::WinUI3;
using namespace ABI::Windows::Data::Json;

namespace AdaptiveCards::Rendering::WinUI3
{
    HRESULT AdaptiveMediaEventArgs::RuntimeClassInitialize() { return S_OK; }

    HRESULT AdaptiveMediaEventArgs::RuntimeClassInitialize(_In_ IAdaptiveMedia* media)
    {
        m_media = media;
        return S_OK;
    }

    HRESULT AdaptiveMediaEventArgs::get_Media(_COM_Outptr_ IAdaptiveMedia** media) { return m_media.CopyTo(media); }
}
